Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -94,23 +94,27 @@ public class PushDeploymentController : ApiController
                         }
                         response.StatusCode = HttpStatusCode.Accepted;
                         break;
-                    case FetchDeploymentRequestResult.ForbiddenScmDisabled:
+                    case FetchDeploymentRequestResultStatus.ForbiddenScmDisabled:
                         // Should never hit this for zip push deploy
                         response.StatusCode = HttpStatusCode.Forbidden;
                         _tracer.Trace("Scm is not enabled, reject all requests.");
                         break;
-                    case FetchDeploymentRequestResult.ConflictAutoSwapOngoing:
+                    case FetchDeploymentRequestResultStatus.ConflictAutoSwapOngoing:
                         response.StatusCode = HttpStatusCode.Conflict;
                         response.Content = new StringContent(Resources.Error_AutoSwapDeploymentOngoing);
                         break;
-                    case FetchDeploymentRequestResult.Pending:
+                    case FetchDeploymentRequestResultStatus.Pending:
                         // Shouldn't happen here, as we disallow deferral for this use case
                         response.StatusCode = HttpStatusCode.Accepted;
                         break;
-                    case FetchDeploymentRequestResult.RanSynchronously:
+                    case FetchDeploymentRequestResultStatus.RanSynchronously:
                         response.StatusCode = HttpStatusCode.OK;
                         break;
-                    case FetchDeploymentRequestResult.ConflictDeploymentInProgress:
+                    case FetchDeploymentRequestResultStatus.RanSynchronouslyFailed:
+                        response.StatusCode = HttpStatusCode.BadRequest;
```

Comment:
I misspoke on this earlier; it should probably be a 500 instead of a 400.

Reply:
Fixed